packages/nextjs/src/config/nextConfigToWebpackPluginConfig.ts

```diff
@@ -0,0 +1,128 @@
+import { NextConfigObject, SentryWebpackPluginOptions } from './types';
+
+/**
+ * About types:
+ * It's not possible to set strong types because they end up forcing you to explicitly
+ * set `undefined` for properties you don't want to include, which is quite
+ * inconvenient. The workaround to this is to relax type requirements at some point,
+ * which means not enforcing types (why have strong typing then?) and still having code
+ * that is hard to read.
+ */
+
+/**
+ * Next.js properties that should modify the webpack plugin properties.
+ * They should have an includer function in the map.
+ */
+export const SUPPORTED_NEXTJS_PROPERTIES = ['distDir'];
+
+type PropIncluderFn = (
+  nextConfig: NextConfigObject,
+  sentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions>,
+) => Partial<SentryWebpackPluginOptions>;
+
+export type PropsIncluderMapType = Record<string, PropIncluderFn>;
+export const PROPS_INCLUDER_MAP: PropsIncluderMapType = {
+  distDir: includeDistDir,
+};
+
+/**
+ * Creates a new Sentry Webpack Plugin config from the given one, including all available
+ * properties in the Nextjs Config.
+ *
+ * @param nextConfig User's Next.js config.
+ * @param sentryWebpackPluginOptions User's Sentry Webpack Plugin config.
+ * @returns New Sentry Webpack Plugin Config.
+ */
+export default function includeAllNextjsProps(
+  nextConfig: NextConfigObject,
+  sentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions>,
+): Partial<SentryWebpackPluginOptions> {
+  return includeNextjsProps(nextConfig, sentryWebpackPluginOptions, PROPS_INCLUDER_MAP, SUPPORTED_NEXTJS_PROPERTIES);
+}
+
+/**
+ * Creates a new Sentry Webpack Plugin config from the given one, and applying the corresponding
+ * modifications to the given next properties. If more than one option generates the same
+ * properties, the values generated last will override previous ones.
+ *
+ * @param nextConfig User's Next.js config.
+ * @param sentryWebpackPluginOptions User's Sentry Webapck Plugin config.
+ * @param nextProps Next.js config's properties that should modify webpack plugin properties.
+ * @returns New Sentry Webpack Plugin config.
+ */
+export function includeNextjsProps(
+  nextConfig: NextConfigObject,
+  sentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions>,
+  propsIncluderMap: Record<string, PropIncluderFn>,
+  nextProps: string[],
+): Partial<SentryWebpackPluginOptions> {
+  // @ts-ignore '__spreadArray' import from tslib, ts(2343)
+  const propsToInclude = [...new Set(nextProps)];
+  return (
+    propsToInclude
+      // Types are not strict enought to ensure there's a function in the map
+      .filter(prop => propsIncluderMap[prop])
+      .map(prop => propsIncluderMap[prop](nextConfig, sentryWebpackPluginOptions))
+      .reduce((prev, current) => ({ ...prev, ...current }), {})
+  );
```

Comment:
Actually now that I'm thinking about it, we prob don't even need `PROPS_INCLUDER_MAP`. We can just iterate through `PROPS_INCLUDER_MAP` and make it the source of truth here. Just use `Object.entries(PROPS_INCLUDER_MAP)`?

Reply:
IIUC, you're suggesting we don't need the argument, right? I think we do, if we want to cover the use case of having a function to include a prop _without_ modifying `PROPS_INCLUDER_PATH`.

On the other hand, about using `Object.entries(`PROPS_INCLUDER_MAP)`, I don't find the value of using it since it's better to iterate through the stuff we want to include, over iterating through the stuff we have and check whether we want to include it (why iterate through stuff we know we don't want, when we know it beforehand?).